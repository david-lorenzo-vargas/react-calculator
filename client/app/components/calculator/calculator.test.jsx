import React from 'react';
import { shallow } from 'enzyme';
import Calculator from './calculator';

describe('Calculator', () => {
  describe('render', () => {
    it('should render corrently', () => {
      const wrapper = shallow(<Calculator />);

      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('state', () => {
    describe('when the component first mounts', () => {
      it('should match the default state', () => {
        const expected = {
          buttonValue: '',
          buttonType: '',
          savedValue: 0,
          savedButtonType: '',
        };
        const wrapper = shallow(<Calculator />);
        const received = wrapper.state();

        expect(received).toMatchObject(expected);
      });
    });
  });

  describe('getTotals', () => {
    describe('when state.buttonType === add', () => {
      it('should sum state.buttonValue and state.savedValue', () => {
        const states = [
          {
            expected: 0,
            state: { buttonValue: 10, savedValue: 10, buttonType: 'substract' },
          },
          {
            expected: 100,
            state: { buttonValue: 10, savedValue: 10, buttonType: 'multiply' },
          },
          {
            expected: 1,
            state: { buttonValue: 10, savedValue: 10, buttonType: 'division' },
          },
          {
            expected: 20,
            state: { buttonValue: 10, savedValue: 10, buttonType: 'add' },
          },
        ];

        states.forEach((stateItem) => {
          const { expected, state } = stateItem;
          const wrapper = shallow(<Calculator />);
          wrapper.setState(state);

          const received = wrapper.instance().getTotals();

          expect(received).toEqual(expected);
        });
      });
    });
  });
});
