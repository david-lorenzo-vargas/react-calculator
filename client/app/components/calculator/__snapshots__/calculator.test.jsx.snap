// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator render should render corrently 1`] = `
<div
  className="calculator"
>
  <Header
    total={0}
  />
  <Body
    onButtonClick={[Function]}
  />
</div>
`;
